### ISSUE TITLE
Saving multiple models that having same custom primary key format will fail

### ISSUE THREAD

**Issue by HuiSF** [aws-amplify/amplify-swift#2832]
### Describe the bug

Original issue report: https://github.com/aws-amplify/amplify-flutter/issues/2519

Take model schema:

```graphql
type TestTableOne @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 

type TestTableTwo @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 

type TestTableThree @model @auth(rules: [{allow: public}]) {
  id: ID!
  profile_id: ID! @primaryKey
  count: Int
} 
```

1. Creating new model instance of each model, with the same value assigned to `profile_id`
2. Saving newly created model instances using `Amplify.DataStore.save` 
3. Only the first record can be saved, and the remaining will encounter exception as the following

```
final profileId = UUID.getUUID();
final testTableOne = TestTableOne(profile_id: profileId, count: 100);
await ddbRepo.saveTestOne(testTableOne);
final testTableTwo = TestTableTwo(profile_id: profileId, count: 101);
await ddbRepo.saveTestTwo(testTableTwo);
final testTableThree = TestTableThree(profile_id: profileId, count: 102);
await ddbRepo.saveTestThree(testTableThree);
```

The 1st save succeeded but the subsequent save failed due to;

```
DataStoreException(message: Received a create mutation for an item that has already been created,
recoverySuggestion: Review your app code and ensure you are not issuing incorrect DataStore.save() calls for
the same model.
```

**Possible Cause**

The logic checking throwing `"Received a create mutation for an item that has already been created"` may only check the primary key field value, without counting model name, so it wrongly determined that the same model is being repeated saved.

The issue exists in amplify-swift v1, the custom primary key implementation is the same in v2, so the issue must exist in v2 as well.

### Steps To Reproduce

```swift
See above details.
```

### Expected behavior

The use case described above should not fail

### Amplify Framework Version

1.38.2

### Amplify Categories

DataStore

### Dependency manager

Cocoapods

### Swift version

5

### CLI version

The latest

### Xcode version

The latest

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Device

iPhone 14 simulators

### iOS Version

iOS 16

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by 5d** [comment]
Hi @HuiSF,
Thanks for reporting. We confirmed the issue exist for both amplify-swift v1 and v2. 
We've made the fix and raised the PRs.

**Comment by 5d** [comment]
This patch has been released in the newly released Version [1.29.2](https://github.com/aws-amplify/amplify-swift/releases/tag/1.29.2).
